internal/devices/update.go

```diff
@@ -80,7 +75,11 @@ func (c *Client) Update() *cobra.Command {
 				deviceUpdate.Userdata = &userdata
 			}
 
-			if locked {
+			if cmd.Flag("locked").Changed {
```

Comment:
A test for lock/unlock would build confidence that we've fixed the issue

Reply:
I don't have a strong preference either way for whether the test hits a live API or a mock.  If we're going to mock I think it's better to set up a mock API than to inject a mock HTTP transport, since that more closely matches real usage, reduces the risk of accidentally mocking out some custom behavior, and provides a more straightforward path to maybe eventually adopting a mock that is generated from the API spec.

Reply:
@ctreatma added in https://github.com/equinix/metal-cli/pull/485/commits/c6562a20d37b122be18c4c5c7f4a10ff1d9310e8